### DISCUSSION TITLE
Quarkus at Red Hat Summit 2023

### DISCUSSION THREAD

**Discussion by insectengine** [quarkusio/quarkus#33126] [Events]
**Event Description:** Join us for Red Hat Summit, which will feature virtual and in-person events that are designed to fit your open source journey.

**Date:**  May 23-25, 2023
Location: Boston, USA
Event Type: In Person

https://www.redhat.com/en/summit

### Quarkus Sessions and Activities

**TUESDAY**

**Quarkus supercharges native Java with GraalVM**
**Date and Time:** Tuesday from 1:10 PM - 1:30 PM EDT
**Session Link:** https://events.experiences.redhat.com/widget/redhat/sum23/SessionCatalog2023/session/1681250505816001iE4Z
**Speaker(s):** Daniel Oh
**Session Type**: Lightning talk
**Abstract:** This lightning talk will demonstrate how Java developers can optimize Java applications with fast startups and response times, and a low memory footprint using GraalVM and Quarkus. 

**Quarkus for Spring Developers**
**Date and Time:** Tuesday from 2:00 PM - 2:20 PM EDT
**Session Link:** https://events.experiences.redhat.com/widget/redhat/sum23/SessionCatalog2023/session/1682460444286001Hgrx
**Speaker(s):** Eric Deandrea
**Session Type:** Lightning Talk
**Abstract:** Want to see how you can run twice as many workloads given the same hardware resources? In this session, I will show concepts and conventions familiar to Spring developers and how those same concepts and conventions can be implemented in Quarkus, all while highlighting similarities and differences between them.

**The future of Java in a Kubernetes world**
**Date and Time:** Tuesday from 3:00 PM - 3:45 PM EDT
**Session Link:** https://events.experiences.redhat.com/widget/redhat/sum23/SessionCatalog2023/session/1673456338993001Ppcw
**Speaker(s):** Mark Little
**Session Type:** Interactive Session
**Abstract:** This session will share the latest innovations in Java, including how Kubernetes-native Java, Quarkus, OpenJDK, and Eclipse Adoptium are proving to be more and more important in a Kubernetes environment, Learn how you can be on the leading edge of Kubernetes-native Java developers and how these innovations are affecting Red Hat products and projects.

**Event-driven autoscaling for Serverless Java**
**Date and Time:** Tuesday from Tuesday, May 235:00 PM - 5:20 PM EDT
**Session Link:** https://events.experiences.redhat.com/widget/redhat/sum23/SessionCatalog2023/session/1682460444677001HfIA
**Speaker(s):** Daniel Oh
**Session Type:** Lightning Talk
**Abstract:** This lightning talk will teach you how to redesign your Kubernetes autoscaling architecture by event-driven metrics from Apache Kafka over standard resources (CPU, Memory) with Knative and KEDA integration for serverless Java using Quarkus.

**WEDNESDAY**

**bp’s journey to success with Red Hat OpenShift Service on AWS**
**Date and Time:** Wednesday from 10:30 AM - 11:15 AM EDT
**Session Link:** https://events.experiences.redhat.com/widget/redhat/sum23/SessionCatalog2023/session/1673432037845001MhoS
**Speaker(s):** Waqaas Kausar & Emma McEwen
**Session Type:** Interactive Session
**Abstract:** In this session, you’ll hear an engaging account of bp’s successful migration from Red Hat OpenShift Container Platform 3.11 to Red Hat OpenShift Service on Amazon Web Services (AWS) for its trading & shipping (T&S) division. bp and Red Hat will discuss why bp chose OpenShift Service on AWS, the migration pilot journey, the results achieved, and post-migration activities.

- Migration of critical applications.
- Integration and storage foundation with Red Hat OpenShift Data Foundation.
- Continuing enablement through Red Hat Services.
- Goals for ongoing migration of all T&S applications.
- New technology capabilities and tools, such as HyperShift and Quarkus.

**Enter serverless functions journey with Quarkus**
**Date and Time:** Wednesday from 11:00 AM - 12:30 PM EDT
**Session Link:** https://events.experiences.redhat.com/widget/redhat/sum23/SessionCatalog2023/session/1671448706376001mamw
**Speaker(s):** Daniel Oh
**Session Type:** Workshop
**Abstract:** This workshop is for developers getting started with scaffolding serverless functions projects using Quarkus, a new Kubernetes-native Java framework with familiar tools like CLI, Maven, and Gradel. This hands-on lab showcases how quickly developers can create cloud-native microservice projects using Quarkus. This workshop will cover:

- Deploying service functions to AWS Lambda, optimizing the functions, and making them portable across multiple serverless platforms—such as AWS Lambda, Azure Function, Google Cloud Platform, or Kubernetes Knative.
- Using the latest command-line tool (Kn func) to enable a Buildpack for function development and deployment in minutes.
- How the application can be deployed to a function to AWS Lambda and Knative Event with JVM and Native mode.

Lab participants will be provided a sandbox for serverless deployments.
Please bring your own laptop to this workshop.

**Accelerate new application development with Red Hat OpenShift**
**Date and Time:** Wednesday from 1:00 PM - 3:00 PM EDT
**Session Link:** https://events.experiences.redhat.com/widget/redhat/sum23/SessionCatalog2023/session/1673369845907001EILW
**Speaker(s):** Camry Fedei & Juliano Mohr
**Session Type:** Lab
**Abstract:** This hands-on workshop will immerse attendees in the latest practices and tooling available to dramatically accelerate development of new applications on Red Hat OpenShift. 

From enabling immediate developer productivity to building a flexible architecture, attendees will learn how to:

- Take advantage of Quarkus capabilities to facilitate agile development and testing.
- Deploy their applications through Red Hat OpenShift GitOps and Red Hat OpenShift Pipelines.
- Design for resilience with an event-driven architecture using Kafka.

Key takeaways include:

- Quarkus development experience with live coding.
- Red Hat OpenShift developer self-provisioning deployment.
- OpenShift Pipelines (Tekton) for continuous integration and delivery.
- OpenShift GitOps (ArgoCD) for continuous deployment.
- Event-driven architecture with Kafka.

**Use Change Data Capture to build new functionality without changing existing code**
**Date and Time:** Wednesday from 1:30 PM - 1:50 PM EDT
**Session Link:** https://events.experiences.redhat.com/widget/redhat/sum23/SessionCatalog2023/session/1682460445045001HCcH
**Speaker(s):** Bernard Tison & Jay Baskaran
**Session Type:** Lightning Talk
**Abstract:** Change data capture (CDC) refers to the process of capturing changes made to data in a database and then delivering those changes in real-time to a downstream process. In this session we will walk you through a demo where CDC is used to build new functionality without changing existing application code. The demo also illustrates other technologies from the Red Hat Application Services Portfolio such as Apache Kafka, Camel and Quarkus. 

**Modern cloud-native Java runtimes performance monitoring on Red Hat OpenShift**
**Date and Time:** Wednesday from 1:00 PM - 3:00 PM EDT
**Session Link:** https://events.experiences.redhat.com/widget/redhat/sum23/SessionCatalog2023/session/1673392884031001ZOQX
**Speaker(s):** Rafael Soares, Giovanni Fontana, & Guilherme Camposo
**Session Type:** Lab
**Abstract:** Java continues to strive in the enterprise and now in cloud environments. Modern Java developers seeking cloud-native experience have various options when building and deploying Java applications in cloud environments.

In this workshop, we'll provide an intro-level, hands-on session on application performance using three different cloud-native Java runtimes: Quarkus, SpringBoot, and Micronaut. We'll be building a simple microservice using these three runtimes, packaging them as containers, and deploying them on top of Red Hat OpenShift. Finally, we'll be analyzing and monitoring CPU load as well as memory usage of each variant using some of the capabilities provided by Red Hat OpenShift Platform, like Prometheus and Grafana. We'll conclude by establishing a baseline, so you can see the impact of scaling out or scaling up containers.

In this session, you'll:

- Create your own GitHub repository based on a GitHub template.
- Develop a microservice that consumes CPU and memory using three different Java Stacks: Quarkus, SpringBoot, and Micronaut.
- Build each microservice as a container.
- Run these microservices locally.
- Deploy to your Red Hat OpenShift project (namespace).
- Execute and monitor using different performance parameters.
- Load test the microservices.
- Analyze CPU load and scale containers appropriately.
- Analyze memory and scale containers appropriately.

**Note:** This is a BYOL (Bring Your Own Laptop) lab.

**Build a multichannel hub with Apache Camel and Quarkus**
**Date and Time:** Wednesday from 3:30 PM - 5:30 PM EDT
**Session Link:** https://events.experiences.redhat.com/widget/redhat/sum23/SessionCatalog2023/session/1673372897061001s5cd
**Speaker(s):** Bruno Meseguer & Eric Deandrea
**Session Type:** Lab
**Abstract:**  The emergence of Quarkus and the latest evolution of Apache Camel has driven Red Hat's application development and enterprise integration to make a significant leap forward, bringing Camel to everyone, not just the Camel developer. This workshop presents a real-world inspired base scenario, where you will:

- Impersonate the regular Kubernetes user as well as the experienced Camel developer.
- Complete a series of exercises, ranging from simple data flows to medium and advanced use cases.
- Take the solution further by extending it with additional Red Hat Runtimes components plugging into the system.

The lab is very easy to follow and delivers great value with each step the student completes. Takeaways for participants will include:

- An insightful view on trends organizations use to connect and collaborate with customers.
- Knowledge of how to build a platform from the ground up with minimal effort.
- Valuable materials to help your customers and colleagues familiarize themselves with new technology.
- A better understanding of how to use the latest generation of Apache Camel.
- Ways to empower the platform with additional services using Red Hat Runtimes.
 

**THURSDAY**

**Modern cloud-native Java runtimes performance monitoring on Red Hat OpenShift**
**Date and Time:** Thursday 10:15 AM - 12:15 PM EDT
**Session Link:** https://events.experiences.redhat.com/widget/redhat/sum23/SessionCatalog2023/session/1673392884031001ZOQX
**Speaker(s):** Rafael Soares, Giovanni Fontana, & Guilherme Camposo
**Session Type:** Lab
**Abstract:** Java continues to strive in the enterprise and now in cloud environments. Modern Java developers seeking cloud-native experience have various options when building and deploying Java applications in cloud environments.

In this workshop, we'll provide an intro-level, hands-on session on application performance using three different cloud-native Java runtimes: Quarkus, SpringBoot, and Micronaut. We'll be building a simple microservice using these three runtimes, packaging them as containers, and deploying them on top of Red Hat OpenShift. Finally, we'll be analyzing and monitoring CPU load as well as memory usage of each variant using some of the capabilities provided by Red Hat OpenShift Platform, like Prometheus and Grafana. We'll conclude by establishing a baseline, so you can see the impact of scaling out or scaling up containers.

In this session, you'll:

- Create your own GitHub repository based on a GitHub template.
- Develop a microservice that consumes CPU and memory using three different Java Stacks: Quarkus, SpringBoot, and Micronaut.
- Build each microservice as a container.
- Run these microservices locally.
- Deploy to your Red Hat OpenShift project (namespace).
- Execute and monitor using different performance parameters.
- Load test the microservices.
- Analyze CPU load and scale containers appropriately.
- Analyze memory and scale containers appropriately.

**Note:** This is a BYOL (Bring Your Own Laptop) lab.

**Comment by mariasalcedo** [comment]
Hi! Thanks for sharing.
I have a question regarding the "virtual" possibilities. 
Is it possible to not go to Boston and see the entirety through streaming?
Thanks in advance!

**Reply by insectengine** [reply]
I know the keynote will be live streamed but I don't know about any of the specific sessions.

**Follow @RedHat on Twitter to watch the livestream of our keynote sessions during Red Hat Summit 2023. Tune in at 9AM ET on May 23 and May 24 to hear the latest from Red Hat executives, customers, and partners.**
